### PR TITLE
Run tests in parallel

### PR DESCRIPTION
**Pull request by bddppq**

### CODE REVIEW

**Comment by bddppq** [comment]
@onnxbot retest this please

**Comment by pietern** [comment]
@bddppq You can also apt get install it here, as it is only used for the tests. The images are already pretty big, but I rather err on the side of not including stuff if it is not strictly necessary.

**Comment by bddppq** [comment]
@pietern Doing apt-get here is not good as it needs sudo and also it's inappropriate in a test script. It's indeed not strictly necessary, that's why this script has the fallback logic when parallel is not available.
I can implement the parallel execution with pure bash here (i.e. without `parallel`). However, in our jenkins CI setup (since we have disabled multi-job execution to avoid indeterministic scheduling) adding parallelism within a single job seems to be good way to increase CPU utilization in general.

**Comment by pietern** [comment]
Agree on the CPU utilization. What about xargs -P ?

**Comment by bddppq** [comment]
@pietern yeah I can implement parallel execution in bash without GNU parallel. However the reason I prefer to use GNU parallel is because it has some nice features (e.g. buffer the output to prevent half a line in the log comes from one process and half a line from another one).
I understand the concern of dragging unnecessary stuffs into the docker images, let's just the `xargs -P` for now and revisit later if we found it's necessary.

File: test.sh [hunk 1]
```diff
@@ -21,8 +42,13 @@ test_files=(
     "$TEST_DIR/test_pytorch_helper.py"
 )
 
-if hash parallel 2>/dev/null; then
-    parallel -j 2 -t --line-buffer --keep-order $PYTHON {} -v ::: "${test_files[@]}"
+if [[ $PARALLEL == 1 ]]; then
+    if hash parallels 2>/dev/null; then
```
Comment (pietern) [inline test.sh]: parallels <-- is this the name? Not `parallel`?

Comment (bddppq) [reply test.sh]: Nice catch, I was trying to go into poor man's version on my dev box but forgot to revert it :)

**Comment by bddppq** [comment]
The test time has deceased from ~12mins to ~8mins :-)

**Comment by pietern** [comment]
Nice!

**Pull request merged**